osu.Game/Screens/Play/Player.cs

```diff
@@ -269,7 +269,12 @@ private void onCompletion()
 
             ValidForResume = false;
 
-            if (!showResults) return;
+            if (!showResults)
+            {
+                // If results shouldn't be shown on completion, just exit
+                this.Exit();
```

Comment:
also i'm unsure if this should be done, but replays just kept looping until you exited yourself so i left it here for now